docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,432 @@
+To increase your program's efficiency, add waypoints such that the path between any two consecutive points can be linear, without intersecting the tissue box or the table:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a start pose, where the cup starts between the gripper's jaws, on the table
+# Start pose has Z of 90mm to grab partway down the 120mm tall cup
+start_pose = Pose(x=320, y=240, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+start_pose_in_frame = PoseInFrame(reference_frame="world", pose=start_pose)
+
+# Create waypoints to increase efficiency of motion planning
+way1_pose = Pose(x=300, y=240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way1_pose_in_frame = PoseInFrame(reference_frame="world", pose=way1_pose)
+way2_pose = Pose(x=300, y=-240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way2_pose_in_frame = PoseInFrame(reference_frame="world", pose=way2_pose)
+
+# Create a pose where the cup will be set down
+end_pose = Pose(x=300, y=-250, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+end_pose_in_frame = PoseInFrame(reference_frame="world", pose=end_pose)
+```
+
+Though we are passing these waypoints in manually, the Motion Service will throw an error if we accidentally pass in a pose that would cause the arm or gripper to hit any obstacles.
+
+Note that the orientations of all the poses are the same.
+If we changed the orientation along the way, we might spill the tea!
+
+{{< alert title="Tip" color="tip" >}}
+You may be wondering how the orientations of the poses are determined.
+Our example gripper's frame is defined such that its orientation vector points from its "wrist" to the tip of its jaws.
+In the example code above, all poses have an [orientation vector](../../../internals/orientation-vector/) pointing along the positive X axis of the world frame, which is a horizontal orientation pointing "forwards" with respect to the xArm 6 base.
+When we tell the gripper to move to such a pose, its orientation vector moves to align with the orientation vector of the pose, so its jaws end up pointing along the global X axis, "forwards" from the robot base.
+This puts it in a good position for picking up and moving a cup.
+
+Additionally, for our gripper, setting `theta=0` about this particular orientation vector orients the gripper such that its jaws open and close horizontally.
+If we changed it to `theta=90` or `theta=270`, the gripper jaws would open vertically, not ideal for picking up a cup!
+{{< /alert >}}
+
+## Add a motion constraint
+
+To keep the cup upright as the arm moves it from one place on the table to another, create a [linear constraint](../../../services/motion/constraints/#linear-constraint).
+Then, when you tell the robot to move the cup from one upright position to another, the gripper will move linearly and the upright orientation of the cup will be maintained throughout the planned path.
```

Comment:
```suggestion
A linear constraint ensures that the gripper will move linearly when you tell the robot to move the cup from one upright position to another, and that it will maintain the upright orientation of the cup throughout the planned path.
```

Reply:
```suggestion
When you tell the robot to move the cup from one upright position to another, a linear constraint forces the gripper to move linearly and to maintain the upright orientation of the cup throughout the planned path.
```